tests/backuptestTriggers.go

```diff
@@ -0,0 +1,634 @@
+	"fmt"
+
+	"math/rand"
+	"time"
+
+	"github.com/pborman/uuid"
+	api "github.com/portworx/px-backup-api/pkg/apis/v1"
+	"github.com/portworx/torpedo/drivers"
+	"github.com/portworx/torpedo/drivers/backup"
+	"github.com/portworx/torpedo/drivers/scheduler"
+
+	"github.com/portworx/torpedo/pkg/log"
+
+	"github.com/onsi/ginkgo"
+	. "github.com/onsi/ginkgo"
+)
+
+// All Longevity types for backup
+
+// Global variables to be used by all flows
+var (
+	LongevityBackupLocationName   string
+	LongevityBackupLocationUID    string
+	LongevityAllNamespaces        []string
+	LongevityClusterUID           string
+	LongevityScheduledAppContexts []*scheduler.Context
+	LongevityAllBackupNames       []string
+)
+
+type PxBackupLongevity struct {
+	CustomData      *CustomData
+	ApplicationData *ApplicationData
+	BackupData      *BackupData
+	RestoreData     *RestoreData
+}
+
+type CustomData struct {
+	Integers map[string]int
+	Strings  map[string]string
+}
+
+type BackupData struct {
+	Namespaces         []string
+	BackupLocationName string
+	BackupLocationUID  string
+	ClusterUid         string
+	BackupName         string
+}
+
+type RestoreData struct {
+	RestoreMap  map[string]string
+	RestoreName string
+}
+
+type ApplicationData struct {
+	SchedulerContext []*scheduler.Context
+}
+
+type EventData struct {
+	SchedulerContext   []*scheduler.Context
+	AppContext         context1.Context
+	BackupNamespaces   []string
+	BackupLocationName string
+	BackupLocationUID  string
+	ClusterUid         string
+	BackupNames        []string
+	RestoreName        string
+}
+
+type EventBuilderResponse struct {
+	Error              error
+	TimeTakenInMinutes float32
+	HighlightEvent     string
+	EventData          EventData
+}
+
+type EventResponse struct {
+	Name               string
+	EventBuilders      map[string]*EventBuilderResponse
+	Errors             []error
+	TimeTakenInMinutes float32
+	HighlightEvents    []string
+	DisruptiveEventRan []string
+	Status             bool
+}
+
+const (
+	EventScheduleApps                   = "EventScheduleApps"
+	EventValidateScheduleApplication    = "EventValidateScheduleApplication"
+	EventAddCredentialandBackupLocation = "EventAddCredentialandBackupLocation"
+	EventAddSourceAndDestinationCluster = "EventAddSourceAndDestinationCluster"
+	EventCreateBackup                   = "EventCreateBackup"
+	EventRestore                        = "EventRestore"
+)
+
+var AllBuilders = map[string]PxBackupEventBuilder{
+	EventScheduleApps:                   eventScheduleApps,
+	EventValidateScheduleApplication:    eventValidateScheduleApplication,
+	EventAddCredentialandBackupLocation: eventAddCredentialandBackupLocation,
+	EventAddSourceAndDestinationCluster: eventAddSourceAndDestinationCluster,
+	EventCreateBackup:                   eventCreateBackup,
+	EventRestore:                        eventRestore,
+}
+
+var ctx, _ = backup.GetAdminCtxFromSecret()
+
+type PxBackupEventBuilder func(*PxBackupLongevity) (error, string, EventData)
```

Comment:
Why do we need a string output here?
Also, error should be always in the end as per convention.

Reply:
String are the highlight events/any anomaly during the event which is not an error.